src/slab.js

```diff
@@ -127,6 +127,9 @@ async function waitForInstance(taskId, taskName) {
       const task_status = body[taskName].status.toLowerCase()
 
       if (task_status === 'done') {
+        if (taskName === 'start') {
+          await acknowledgeTaskDone(taskId)
+        }
```

Comment:
so the process is:

ask for start -> slab starts instance, starts watchdog -> action waits for instance -> acknowledge ?

isn't there a risk of slab killing the instance before the action can aknowledge ?

Reply:
also is there a risk of aknowledging before Slab is waiting for an aknwoledge ?

Reply:
> isn't there a risk of slab killing the instance before the action can aknowledge ?

Reasonably, no. The sleeping time before killing the instance is 5 minutes on Slab side, AFTER the spawn is over.


> also is there a risk of aknowledging before Slab is waiting for an aknwoledge ?

No this is enforced on Slab side, you cannot acknowledge an InProgress or a Failed task (it's basically a no-op)

Reply:
is it possible for the action to acknowledge an in progress task ?

and if so, can it say "ok I acknowledged, now I do something else" and never acknowledges again, meaning slab could kill the instance ?

Reply:
Nope it's not possible to acknowledge an in progress task. The implementation imposes the acknowledge call to happen after a Done task has been received.